pkg/apis/keycloak/v1alpha1/keycloakclient_types.go

```diff
@@ -164,6 +164,9 @@ type KeycloakAPIClient struct {
 	// Authorization settings for this resource server.
 	// +optional
 	AuthorizationSettings *KeycloakResourceServer `json:"authorizationSettings,omitempty"`
+	// Service account roles for this client.
+	// +optional
+	ServiceAccountRoles []string `json:"serviceAccountRoles,omitempty"`
```

Comment:
I would not place this in `KeycloakAPIClient` as it represents the Client in the Keycloak REST API. And the REST API doesn't have any `serviceAccountRoles` field. I think it should be placed in `KeycloakClientSpec`.